packages/lib/checkRateLimitAndThrowError.ts

```diff
@@ -1,4 +1,4 @@
-import { TRPCError } from "@calcom/trpc";
+import { TRPCError } from "@calcom/trpc/server";
```

Comment:
!!!IMPORTANT!!!

We needed to swap this TRPC import from client side to server side because the AI app uses it server side.
Rate limiting in our use case **Does not work** without this.

We don't know if this causes breaking changes across the rest of the site.
We **need a senior engineer on Cal team** to verify that this change is okay.

We don't know if this has upstream repercussions.

Please don't merge this without checking this out. 🙏

Reply:
@zomars @emrysal @PeerRich 

Flagging this, otherwise, this PR should be good to 🚀

Reply:
This looks safe to me <3 

This is only ever ran on the server in all of use cases.